### PR TITLE
[azure] [feat] Implement update_tag, delete_tag of a resource

### PR DESCRIPTION
**Pull request by 1101-1**
# Description

## Added Features

* New Methods:
   * `update_resource_tag`: This method allows for updating a specific tag associated with a resource by its tag name and tag value.
   * `delete_resource_tag`: This method allows for deleting a specific tag associated with a resource by its tag name.

## How it works

* `update_resource_tag`:
    * Pass the tag name and the new tag value to update the respective tag on the specified resource.

* `delete_resource_tag`:
    * Provide the tag name to delete the corresponding tag from the specified resource.

# To-Dos

- [ ] Add test coverage for new or updated functionality
- [x] Lint and test with `tox`
- [x] Implement an `update_tag` and `delete_tag` method in AzureClient
- [x] Implement `update_tag` and `delete_tag` in AzureResource by retrieving the client and calling `update_tag` method.

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://resoto.com/code-of-conduct).

### CODE REVIEW

File: plugins/azure/resoto_plugin_azure/azure_client.py [hunk 1]
```diff
@@ -81,6 +90,62 @@ def delete(self, resource_id: str) -> bool:
...
+            resource = self.client.resources.get_by_id(resource_id=resource_id, api_version="2021-04-01")
+
+            # Check if the tag already exists in the resource's tags
+            existing_tag_value = resource.tags.get(tag_name)
+
+            # If the tag exists, update its value
+            if existing_tag_value is not None:
+                resource.tags[tag_name] = tag_value
+            else:
+                return False  # Tag not found in the resource
```
Comment (aquamatthias) [inline plugins/azure/resoto_plugin_azure/azure_client.py:105]: In case the tag doesn't exist, you should create it.

**Review by aquamatthias — COMMENTED** [review]
Thanks. I added a couple of proposals to improve the code.

File: plugins/azure/resoto_plugin_azure/azure_client.py [hunk 2]
```diff
@@ -81,6 +90,62 @@ def delete(self, resource_id: str) -> bool:
...
+
+        except HttpResponseError as e:
+            if e.error and e.error.code == "ResourceNotFoundError":
+                return False  # Resource not found
+            elif e.error and e.error.code == "ResourceExistsError":
+                return False  # Tag for deletion does not exist
+            else:
+                raise e
+
+        return True
```
[the remark this answers is not in the document]
Comment (1101-1) [reply plugins/azure/resoto_plugin_azure/azure_client.py:132]: I will try to do this.

**Review by aquamatthias — APPROVED** [review]
Nice!

**Pull request merged**